LGM crashes on startup

I installed LGM + enigma from my AUR package [here](https://aur.archlinux.org/packages/enigma-dev-git/).
It used to work fine, but now LGM crashes right after starting (the splash goes away and a window briefly appears):

```
Java Version: 10700 (1.7.0_79)
Loading lib files in /opt/enigma-dev-git/lateralgm.jar
01_move.lgl 02_main1.lgl 03_main2.lgl 04_control.lgl 
05_score.lgl 06_extra.lgl 07_draw.lgl 
Calling `make eTCpath=""`
> make -j 3 -C CompilerSource 
> make[1]: Entering directory '/opt/enigma-dev-git/CompilerSource'
> make[1]: Nothing to be done for 'all'.
> make[1]: Leaving directory '/opt/enigma-dev-git/CompilerSource'
0
Process terminated
Initializing Enigma: 
Failed to create make directory at /home/gandalf3/.enigma/
Make directory is: "/home/gandalf3/.enigma/"
Linking up to IDE
Implementing JDI basics
UnimplementedUnimplemented

Choosing language: C++
Reading GCC builtins
Read key `defines` as `cpp -dM -x c++ -E $blank`
Parsed `cpp` `-dM -x c++ -E "/home/gandalf3/.enigma/enigma_blank.txt"`: redirect=yes
TRUE




*********** EXECUTE 
`/usr/bin/cpp`
`-dM`
`-x`
`c++`
`-E`
`/home/gandalf3/.enigma/enigma_blank.txt`



Call succeeded
Read key `searchdirs` as `gcc -E -x c++ -v $blank`
Parsed `gcc` `-E -x c++ -v "/home/gandalf3/.enigma/enigma_blank.txt"`: redirect=yes
TRUE




*********** EXECUTE 
`/usr/bin/gcc`
`-E`
`-x`
`c++`
`-v`
`/home/gandalf3/.enigma/enigma_blank.txt`



Call succeeded
Searching for directories between "#include <...> search starts here:" and "End of search list."
Toolchain returned 9 search directories:
Creating parse context
Parsing settings...
TOINT: 0
TOINT: 0
TOINT: 1
TOINT: 0
TOINT: 0
TOINT: 0
Setting up IDE editables... 
java: symbol lookup error: /opt/enigma-dev-git/libcompileEGMf.so: undefined symbol: _Z2fcPKc
```

I've tried running it with java 8, but the same thing happens.

@blurymind  I installed [bin32-jre](https://aur.archlinux.org/packages/bin32-jre/) from the AUR and ran `java32 -jar lateralgm.jar` in order to use it.

@RobertBColton I tried deleting `libcompileEGMf.so`, but got the same error. Full output:

    % java32 -jar lateralgm.jar
	Java Version: 10800 (1.8.0_66)
	Loading lib files in /opt/enigma-dev-git/lateralgm.jar
	01_move.lgl 02_main1.lgl 03_main2.lgl 04_control.lgl 
	 05_score.lgl 06_extra.lgl 07_draw.lgl 
	Calling `make eTCpath=""`
	> make -j 3 -C CompilerSource 
	> make[1]: Entering directory '/opt/enigma-dev-git/CompilerSource'
	> g++ -shared -o ../libcompileEGMf.so .eobjs/./syntax/checkfile.o .eobjs/./syntax/syntax.o .eobjs/./settings-parse/crawler.o .eobjs/./settings-parse/eyaml.o .eobjs/./settings-parse/parse_ide_settings.o .eobjs/./parser/collect_variables.o .eobjs/./parser/object_storage.o .eobjs/./parser/parser.o .eobjs/./parser/parser_components.o .eobjs/./languages/lang_CPP.o .eobjs/./languages/language_adapter.o .eobjs/./general/bettersystem.o .eobjs/./general/darray.o .eobjs/./general/macro_integration.o .eobjs/./general/string.o .eobjs/./gcc_interface/gcc_backend.o .eobjs/./filesystem/file_find.o .eobjs/./compiler/reshandlers/rectpack.o .eobjs/./compiler/reshandlers/refont.o .eobjs/./compiler/pcs/pcs.o .eobjs/./compiler/event_reader/event_parser.o .eobjs/./compiler/components/handle_templates.o .eobjs/./compiler/components/module_write_backgrounds.o .eobjs/./compiler/components/module_write_fonts.o .eobjs/./compiler/components/module_write_paths.o .eobjs/./compiler/components/module_write_sounds.o .eobjs/./compiler/components/module_write_sprites.o .eobjs/./compiler/components/parse_and_link.o .eobjs/./compiler/components/parse_secondary.o .eobjs/./compiler/components/write_defragged_events.o .eobjs/./compiler/components/write_event_code.o .eobjs/./compiler/components/write_font_info.o .eobjs/./compiler/components/write_globals.o .eobjs/./compiler/components/write_object_access.o .eobjs/./compiler/components/write_object_data.o .eobjs/./compiler/components/write_room_data.o .eobjs/./compiler/components/write_shader_data.o .eobjs/./compiler/compile.o .eobjs/./compiler/compile_common.o .eobjs/./compiler/jdi_utility.o .eobjs/./compiler/output_locals.o .eobjs/./backend/JavaCallbacks.o .eobjs/./backend/ideprint.o .eobjs/./JDI/test/debug_lexer.o .eobjs/./JDI/src/System/builtins.o .eobjs/./JDI/src/System/lex_buffer.o .eobjs/./JDI/src/System/lex_cpp.o .eobjs/./JDI/src/System/macros.o .eobjs/./JDI/src/System/symbols.o .eobjs/./JDI/src/System/token.o .eobjs/./JDI/src/Storage/definition.o .eobjs/./JDI/src/Storage/full_type.o .eobjs/./JDI/src/Storage/references.o .eobjs/./JDI/src/Storage/value.o .eobjs/./JDI/src/Storage/value_funcs.o .eobjs/./JDI/src/Parser/readers/read_expression.o .eobjs/./JDI/src/Parser/readers/read_next_token.o .eobjs/./JDI/src/Parser/readers/read_operatorkw_name.o .eobjs/./JDI/src/Parser/readers/read_qualified_definition.o .eobjs/./JDI/src/Parser/readers/read_template_parameters.o .eobjs/./JDI/src/Parser/readers/read_type.o .eobjs/./JDI/src/Parser/handlers/handle_class.o .eobjs/./JDI/src/Parser/handlers/handle_declarators.o .eobjs/./JDI/src/Parser/handlers/handle_enum.o .eobjs/./JDI/src/Parser/handlers/handle_function_impl.o .eobjs/./JDI/src/Parser/handlers/handle_hypothetical.o .eobjs/./JDI/src/Parser/handlers/handle_namespace.o .eobjs/./JDI/src/Parser/handlers/handle_scope.o .eobjs/./JDI/src/Parser/handlers/handle_templates.o .eobjs/./JDI/src/Parser/handlers/handle_union.o .eobjs/./JDI/src/Parser/base.o .eobjs/./JDI/src/Parser/parse_context.o .eobjs/./JDI/src/General/debug_macros.o .eobjs/./JDI/src/General/llreader.o .eobjs/./JDI/src/General/parse_basics.o .eobjs/./JDI/src/General/svg_simple.o .eobjs/./JDI/src/API/AST.o .eobjs/./JDI/src/API/AST_Export.o .eobjs/./JDI/src/API/context.o .eobjs/./JDI/src/API/error_reporting.o .eobjs/./JDI/src/API/jdi.o .eobjs/./JDI/src/API/lexer_interface.o .eobjs/./JDI/src/API/user_tokens.o .eobjs/./frontend.o .eobjs/./main.o .eobjs/./makedir.o .eobjs/./settings.o 
	> /usr/bin/ld: cannot open output file ../libcompileEGMf.so: Permission denied
	> collect2: error: ld returned 1 exit status
	> Makefile:55: recipe for target '../libcompileEGMf.so' failed
	> make[1]: Leaving directory '/opt/enigma-dev-git/CompilerSource'
	> make[1]: *** [../libcompileEGMf.so] Error 1
	> make: *** [ENIGMA] Error 2
	> Makefile:6: recipe for target 'ENIGMA' failed
	2
	Process terminated
	Thread-4: 
	java.lang.UnsatisfiedLinkError: Unable to load library 'compileEGMf': Native library (linux-x86/libcompileEGMf.so) not found in resource path ([file:/opt/enigma-dev-git/plugins/enigma.jar])
	        at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:271)
	        at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)
	        at com.sun.jna.Library$Handler.<init>(Library.java:147)
	        at com.sun.jna.Native.loadLibrary(Native.java:412)
	        at com.sun.jna.Native.loadLibrary(Native.java:391)
	        at org.enigma.EnigmaRunner.attemptLib(EnigmaRunner.java:247)
	        at org.enigma.EnigmaRunner.access$1(EnigmaRunner.java:240)
	        at org.enigma.EnigmaRunner$3.run(EnigmaRunner.java:178)
	java.lang.UnsatisfiedLinkError: Unable to load library 'compileEGMf': Native library (linux-x86/libcompileEGMf.so) not found in resource path ([file:/opt/enigma-dev-git/plugins/enigma.jar])
	        at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:271)
	        at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)
	        at com.sun.jna.Library$Handler.<init>(Library.java:147)
	        at com.sun.jna.Native.loadLibrary(Native.java:412)
	        at com.sun.jna.Native.loadLibrary(Native.java:391)
	        at org.enigma.EnigmaRunner.attemptLib(EnigmaRunner.java:247)
	        at org.enigma.EnigmaRunner.access$1(EnigmaRunner.java:240)
	        at org.enigma.EnigmaRunner$3.run(EnigmaRunner.java:178)

--- a/CompilerSource/settings-parse/parse_ide_settings.cpp
+++ b/CompilerSource/settings-parse/parse_ide_settings.cpp
@@ -47,7 +47,7 @@ using namespace std;
 
 string makedir = "";
 
-string fc(const char* fn);
+inline string fc(const char* fn);
 static void clear_ide_editables()
 {
   ofstream wto;

